evalml/pipelines/components/transformers/encoders/ordinal_encoder.py

```diff
@@ -0,0 +1,322 @@
+            pd.DataFrame: Transformed data, where each ordinal feature has been encoded into
+            a numerical column using where of ordinal integers represent
+            the relative order of categories.
+        """
+        X = infer_feature_types(X)
+
+        X_copy = X.ww.copy()
+        X = X.ww.drop(columns=self.features_to_encode)
+
+        # Call sklearn's transform on only the ordinal columns
+        if len(self.features_to_encode) > 0:
+            X_ord = pd.DataFrame(
+                self._encoder.transform(X_copy[self.features_to_encode]),
+                index=X_copy.index,
+            )
+            X_ord.columns = self._get_feature_names()
+            X_ord.ww.init(logical_types={c: "Double" for c in X_ord.columns})
+            self._feature_names = X_ord.columns
+
+            X = ww.utils.concat_columns([X, X_ord])
+
+        return X
+
+    def _get_feature_names(self):
+        """Return feature names for the ordinal features after fitting.
+
+        Since ordinal encoding creates one encoded feature per column in features_to_encode, feature
+        names are formatted as {column_name}_ordinally_encoded --> choose a better name?? maybe one that includes how many categories were encoded
+
+        Returns:
+            np.ndarray: The feature names after encoding, provided in the same order as input_features.
+        """
+        self._features_to_drop = []
+        unique_names = []
+        provenance = {}
+        for col_name in self.features_to_encode:
+            encoded_name = f"{col_name}_ordinally_encoded"
+            unique_names.append(encoded_name)
+            provenance[col_name] = [encoded_name]
+        self._provenance = provenance
+        return unique_names
+
+    def categories(self, feature_name):
+        """Returns a list of the unique categories to be encoded for the particular feature, in order.
+
+        Args:
+            feature_name (str): The name of any feature provided to ordinal encoder during fit.
+
+        Returns:
+            np.ndarray: The unique categories, in the same dtype as they were provided during fit.
+
+        Raises:
+            ValueError: If feature was not provided to ordinal encoder as a training feature.
+        """
+        try:
+            index = self.features_to_encode.index(feature_name)
+        except Exception:
+            raise ValueError(
+                f'Feature "{feature_name}" was not provided to ordinal encoder as a training feature',
+            )
+        return self._encoder.categories_[index]
+
+    def get_feature_names(self):
+        """Return feature names for the ordinal features after fitting.
+
+        Feature names are formatted as {column name}_ordinally_encoded.
+
+        Returns:
+            np.ndarray: The feature names after encoding, provided in the same order as input_features.
+        """
+        return self._get_feature_names()
+
+    def _get_feature_provenance(self):
+        return self._provenance
```

Comment:
This isn't yet covered by tests, I assume, because it's not used in the EvalML pipeline. I didn't see tests in the other components for this method, so are we okay leaving this uncovered for now?

Reply:
I'm certainly not fussed about it, but if anyone disagrees speak now